src/js/port_handler.js

```diff
@@ -321,10 +342,35 @@ PortHandler.updatePortSelect = function (ports) {
         this.addNoPortSelection();
     }
 
+    if (isWeb()) {
+        const askpermission = 'askpermission';
+        this.portPickerElement.append($("<option/>", {
+            value: askpermission,
+            text: i18n.getMessage('portsSelectPermission'),
+            /**
+             * @deprecated please avoid using `isDFU` and friends for new code.
+             */
+            //data: {isShowPermission: true},
+        }));
+
+        this.portPickerElement.on("change", e => {
+            if (e.target.value === askpermission) {
+                this.askPermissionPort();
+            }
+            e.target.value = e.target.options[0].value;
+        });
```

Comment:
In the nutshell, instead of retrieving the reference to input selector, we'd need to use `evenBus` in that element. When values changes we would `$emit` in vuew component.

Then in here instead of doing `this.portPickerElement.on` we would have `evnetBus.$on` here and bind it to this logic.

While a bit more verbose, we get cleaner separation between vue and jquery that way.

Reply:
I was sure that this is not the correct way of getting the element, but I didn't know how to do it properly. I will try according to your instructions.

Reply:
I modified the code to use the bus. I think it's ok now.